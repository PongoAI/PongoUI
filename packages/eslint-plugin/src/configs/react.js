/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
