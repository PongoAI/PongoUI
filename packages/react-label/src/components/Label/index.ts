export * from './Label';
export * from './useLabelStyles';
