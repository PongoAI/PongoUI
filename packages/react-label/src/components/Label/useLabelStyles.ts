import { makeStyles, mergeClasses } from '@griffel/react';
import { tokens } from '@pongo-ui/react-theme';
import type { LabelState } from '@fluentui/react-label';

const useRootStyles = makeStyles({
  root: {
    fontFamily: tokens.fontFamilyBase,
  },

  disabled: {
    color: tokens.inheritDisabled,
  },

  enabled: {
    color: tokens.textColor,
  },

  small: {
    fontSize: tokens.fontSize200,
    lineHeight: tokens.lineHeight200,
  },

  medium: {
    fontSize: tokens.fontSize300,
    lineHeight: tokens.lineHeight300,
  },

  large: {
    fontSize: tokens.fontSize400,
    lineHeight: tokens.lineHeight400,
    fontWeight: tokens.medium,
  },

  strong: {
    fontWeight: tokens.bold,
  },
});

const useRequiredStyles = makeStyles({
  required: {
    paddingLeft: '4px',
  },
  enabled: {
    color: tokens.danger,
  },
  disabled: {
    color: tokens.dangerDisabled,
  },
});

export const useLabelStyles = (state: LabelState): LabelState => {
  const rootStyles = useRootStyles();
  const requiredStyles = useRequiredStyles();

  state.root.className = mergeClasses(
    rootStyles.root,
    state.disabled ? rootStyles.disabled : rootStyles.enabled,
    rootStyles[state.size],
    state.strong && rootStyles.strong,
    state.root.className,
  );

  if (state.required) {
    state.required.className = mergeClasses(
      requiredStyles.required,
      state.disabled ? requiredStyles.disabled : requiredStyles.enabled,
      state.required.className,
    );
  }

  return state;
};
