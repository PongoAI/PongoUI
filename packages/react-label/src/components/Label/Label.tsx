import * as React from 'react';
import { useLabel_unstable, renderLabel_unstable } from '@fluentui/react-label';
import { useLabelStyles } from './useLabelStyles';
import type { LabelProps } from '@fluentui/react-label';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The Label control provides a title to a form component.
 */
export const Label: ForwardRefComponent<LabelProps> = React.forwardRef((props, ref) => {
  const state = useLabel_unstable(props, ref);

  useLabelStyles(state);

  return renderLabel_unstable(state);
});
Label.displayName = 'Label';
