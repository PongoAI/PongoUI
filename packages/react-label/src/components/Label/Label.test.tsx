import React from 'react';
import { Label } from './Label';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';

describe('Label', () => {
  isConformant({
    Component: Label,
    displayName: 'Label',
  });

  it('renders a basic label', () => {
    const renderedComponent = render(<Label>This is a test</Label>);

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
