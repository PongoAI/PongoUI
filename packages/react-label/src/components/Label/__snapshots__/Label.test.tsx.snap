// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders a basic label 1`] = `
<body>
  <div>
    <label
      class="___1purmq9 f8dqhol fc467gg f1arxnym f1e56gn6"
    >
      This is a test
    </label>
  </div>
</body>
`;
